### DISCUSSION TITLE
Bezpieczeństwo danych (login i hasło)

### DISCUSSION THREAD

**Discussion by posadzka** [PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus#163] [General]
Cześć,

Jak wygląda sprawa z bezpieczeństwem loginu i hasła do elicznika? Skonfigurowałem sobie HA+HACS+Tauron-AMIplus, jednak mam obawy przed podaniem swoich danych logowania do elicznika. Jaką mam gwarancję, że ktoś nie zaloguje się na moje konto? Przejrzałem dyskusje i Twoją dokumentację ale nigdzie nic nie znalazłem na ten temat.
Mógłbyś trochę opisać jak wygląda przechowywanie, wykorzystanie i dostęp do danych logowania? Z punktu widzenia HA/HACS i Twojej integracji?
Pozdrawiam
Jurek

**Comment by PiotrMachowski** [comment]
Hej,

Dane są tak bezpieczne, jak bezpieczne jest Twoje HA. Nie wychodzą nigdzie poza Twój serwer (zakładając, że nikomu nie wyślesz np. swojego backupu). Trzymane są w plaintexcie w pliku `config/.storage/core.config_entries` (czyli tak samo, jak dla wszystkich innych integracji). Wykorzystywane są tylko i wyłącznie do pobierania danych z eLicznika, cały kod integracji jest publicznie dostępny, także jeśli masz ochotę, to możesz sam sprawdzić, że nie ma tam nic podejrzanego.

Jedyne niebezpieczeństwo potencjalnie jakie widzę w samej integracji, to ewentualna blokada konta na 8h ze względu na częste logowanie (jeśli np. kilka razy pod rząd zrestartujesz HA). Zawsze istnieje możliwość, że kiedyś Tauron będzie bardziej niezadowolony z powodu automatycznego logowania i pobierania danych, ale na ten moment nie wydaje się to być bardzo prawdopodobne.
